src/Authenticator/JwtAuthenticator.php

```diff
@@ -89,8 +90,8 @@ public function authenticate(ServerRequestInterface $request): ResultInterface
 
         $result = json_decode(json_encode($result), true);
 
-        $key = IdentifierInterface::CREDENTIAL_JWT_SUBJECT;
-        if (empty($result[$key])) {
+        $dataKey = $this->getConfig('dataField');
```

Comment:
I was suggesting `dataKey` for the option name, not just the variable name :slightly_smiling_face:.

Reply:
pfff, ok. but this would break backward compatibility? IF anyone have already defined this with the default value of sub, when he will change the value this will not work anymore.

Reply:
Don't really understand what backwards compatibility concern you have. We are adding a new, so whether it's named `dataField` or `dataKey` really doesn't matter.

That's said let's see if anyone else has other suggestions regarding the new key name.

Reply:
you are right. it looks its too much for today. I have looking at Authentication.JwtSubject config and have messed up. Yes, thats new config. The only concern that is is that all other components have dataField naming. If thats ok I can change to dataKey.

Reply:
> The only concern that is is that all other components have dataField naming. If thats ok I can change to dataKey.

While naming consistency is important it should be contextual too. Personally I haven't seem the term "field" used in context of JWT. In JWT parlance the `sub` key is a ["claim"](https://tools.ietf.org/html/draft-ietf-oauth-json-web-token-27#section-4.1.2). So perhaps something more verbose like `identityClaimKey` might be ever better.

Reply:
I agree. Maybe `subjectClaimKey`

Reply:
`claimKey` or `subjectKey` are shorter names that are inline with other JWT jargon.

Reply:
It would be good to add a test for this configuration option to help prevent regressions in the future.